source/common/tracing/zipkin/span_buffer.cc

```diff
@@ -2,7 +2,7 @@
 
 namespace Zipkin {
 
-bool SpanBuffer::addSpan(SpanPtr&& span) {
+bool SpanBuffer::addSpan(Span&& span) {
```

Comment:
Why are we moving away from passing pointers here? I'm fairly certain that if you define a copy constructor on Span (which you have) and you don't define an explicit move constructor, you are actually going to be copying. This is much less efficient. You also have to run destructors in the buffer array every time you clear. I strongly suspect that you want to revert these changes, but not totally sure why they were made. Can potentially help with design if there is more context.